sqlmesh/core/macros.py

```diff
@@ -285,6 +288,27 @@ def columns_to_types(self, model_name: str) -> t.Dict[str, exp.DataType]:
 
         return columns_to_types
 
+    def resolve_tables(self, query: exp.Expression, **kwargs) -> exp.Expression:
+        """Resolves queries with references to SQLMesh model names to their physical tables."""
+        return self.locals["resolve_tables"](query, **kwargs)
+
+    @property
+    def runtime_stage(self) -> RuntimeStage:
+        """Returns the current runtime stage of the macro evaluation."""
+        return self.locals["runtime_stage"]
+
+    @property
+    def engine_adapter(self) -> EngineAdapter:
+        engine_adapter = self.locals.get("engine_adapter")
+        if not engine_adapter:
+            raise SQLMeshError(
+                "The engine adapter is not available while models are loading."
+                " Rendering queries happens frequently and should be fast and determinstic so it is not recommended"
+                " to make calls to the warehouse at load time."
+                " You can gate these calls by checking in Python: evaluator.runtime_stage != 'loading' or SQL: @runtime_stage <> 'loading'."
```

Comment:
It's not like this is a recommendation, since we're prohibiting the use of this property at load time, so perhaps the reasoning behind why this can't happen should go in the docs instead? I think I'd personally keep only the first and last sentences here.